FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift

```diff
@@ -800,7 +798,160 @@ class FRAClientTests: FRABaseTests {
+           ex2.fulfill()
+        })
+        waitForExpectations(timeout: 60, handler: nil)
+        
+        XCTAssertEqual(authenticatorManager.getAllAccounts().count, 2)
+        XCTAssertNotNil(authenticatorManager.getAccount(identifier: "ForgeRockSandbox-pushtestuser"))
+        XCTAssertEqual(authenticatorManager.getAccount(identifier: "ForgeRockSandbox-pushtestuser")?.mechanisms.count, 1)
+        XCTAssertNotNil(authenticatorManager.getAccount(identifier: "ForgeRock-pushtestuser2"))
+        XCTAssertEqual(authenticatorManager.getAccount(identifier: "ForgeRock-pushtestuser2")?.mechanisms.count, 1)
+        
+        XCTAssertEqual(storageClient.defaultStorageClient.accountStorage.allItems()?.count, 2)
+        XCTAssertEqual(storageClient.defaultStorageClient.mechanismStorage.allItems()?.count, 2)
+        XCTAssertEqual(storageClient.defaultStorageClient.notificationStorage.allItems()?.count, 0)
+
+
+        //  Make sure to mimic MechanismUUID
+        guard let account = authenticatorManager.getAccount(identifier: "ForgeRockSandbox-pushtestuser"),
+           let mechanism = account.mechanisms.first as? PushMechanism else {
+           XCTFail("Failed to retrieve PushMechanism")
+           return
+        }
+        //  Update Mechanism with UUID
+        mechanism.mechanismUUID = "759ACE9D-C64B-43E6-981D-97F7B54C3B01"
+        XCTAssertTrue(FRAClient.storage.setMechanism(mechanism: mechanism))
+
+        guard let account2 = authenticatorManager.getAccount(identifier: "ForgeRock-pushtestuser2"),
+           let mechanism2 = account2.mechanisms.first as? PushMechanism else {
+           XCTFail("Failed to retrieve PushMechanism")
+           return
+        }
+        //  Update Mechanism with UUID
+        mechanism2.mechanismUUID = "862BCE9A-A74A-33D2-181D-77F6B54C3A10"
+        XCTAssertTrue(FRAClient.storage.setMechanism(mechanism: mechanism2))
+        
+        //  Fake first remote-message receive action
+        var payload: [String: Any] = [:]
+        var aps: [String: Any] = [:]
+        aps["messageId"] = "AUTHENTICATE:64e909a2-84db-4ee8-b244-f0dbbeb8b0ff1589151035455"
+        aps["content-available"] = true
+        aps["alert"] = "Login attempt from user at ForgeRockSandbox"
+        aps["data"] = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJjIjoicGtJZm1rZDFSbDNJWnFhSmNZSUhLUzJic2wvZnhiWGNPYTEvNE83Z2pMTT0iLCJ0IjoiMTIwIiwidSI6Ijc1OUFDRTlELUM2NEItNDNFNi05ODFELTk3RjdCNTRDM0IwMSIsImwiOiJZVzFzWW1OdmIydHBaVDB3TVE9PSJ9.o_F0jvSiQlcpInyexi5ED4pjKNHxoHknPIsowYbpyYQ"
+        aps["sound"] = "default"
+
+        payload["aps"] = aps
+
+        guard let _ = FRAPushHandler.shared.application(UIApplication.shared, didReceiveRemoteNotification: payload) else {
+          XCTFail("Failed to parse notification payload and construct PushNotification object")
+          return
+        }
+
+        var pushAccount = authenticatorManager.getAccount(identifier: "ForgeRockSandbox-pushtestuser")
```

Comment:
maybe reuse the Account instance retrieved from line#852?

Reply:
Hmm.. I guess that's what I was doing in other tests.. it's may bad.

Reply:
Yes, I reused the previous```test_07_get_all_notifications_for_mechanism``` as a baseline. The reason it's needed is the following Assert:

`XCTAssertEqual(pushMechanism?.notifications.count, 1)`

The ```notifications``` attribute under PushMechanism is only initialized when an account is retrieved using `getAccount()` or `getAllAccounts()`.

Reply:
I see, that makes sense. Perhaps it's something that we may want to discuss further on how/when we should read it from the storage, but for this story itself, I believe that it should be okay.

Thanks,